### PR TITLE
Fix one way tests can hang

### PR DESCRIPTION
**Pull request by soltanmm**
I've witnessed at least one hang in GDB being that all threads have finished save for a pipe thread. This gets *that* particular instance.

But it appears there's still another flaky hang somewhere in the tests that occurs with some 250 threads running. :-(

### CODE REVIEW

**Comment by nathanielmanistaatgoogle** [comment]
Looks good; merge when test results land.

**Comment by soltanmm** [comment]
Updated; now it fixes *two* ways. Two's better than one, right?

File: src/python/grpcio/tests/_runner.py [hunk 1]
```diff
@@ -43,6 +43,8 @@
 from tests import _loader
 from tests import _result
 
+_READ_BYTES = 2**20
```
Comment (nathanielmanistaatgoogle) [inline src/python/grpcio/tests/_runner.py]: How about a comment on this constant justifying its existence by explaining everything we've learned today?

**Comment by nathanielmanistaatgoogle** [comment]
Your commit log message will need to be edited.

File: src/python/grpcio/tests/_runner.py [hunk 2]
```diff
@@ -76,6 +78,7 @@ def start(self):
     flags = fcntl.fcntl(self._read_fd, fcntl.F_GETFL)
     fcntl.fcntl(self._read_fd, fcntl.F_SETFL, flags | os.O_NONBLOCK)
     self._read_thread = threading.Thread(target=self._read)
+    self._read_thread.daemon = True
```
Comment (nathanielmanistaatgoogle) [inline src/python/grpcio/tests/_runner.py]: Perhaps a comment here too?

**Comment by nathanielmanistaatgoogle** [comment]
Okay, so it turns out you _were_ editing your commit log message...

**Comment by soltanmm** [comment]
Updated.

**Comment by jtattermusch** [comment]
@soltanmm, Good job tracking this one down!

**Comment by soltanmm** [comment]
Python (mercifully) green.

**Pull request merged**